src/Datadog.Trace/Agent/ExporterWriter.cs

```diff
@@ -42,16 +42,13 @@ public void WriteTrace(Span[] trace)
                 Log.Warning("Trace buffer is full. Dropping a trace from the buffer.");
             }
 
-            if (_statsd != null)
-            {
-                _statsd.Increment(TracerMetricNames.Queue.EnqueuedTraces);
-                _statsd.Increment(TracerMetricNames.Queue.EnqueuedSpans, trace.Length);
+            _metrics.Increment(TracerMetricNames.Queue.EnqueuedTraces);
```

Comment:
This can be solved a lot easier by just adding '?' operator instead of if:
_statsd?.Increment(TracerMetricNames.Queue.EnqueuedTraces);
but I understand this is just a perk of having separate interface for metrics, that is mostly there for decoupling reasons.

Reply:
Yeah. I do not think we want to couple our code to `DogStatsd` 😉